source/pele/xy_model_online.h

```diff
@@ -0,0 +1,232 @@
+#ifndef _PELE_XY_MODEL_ONLINE_H
+#define _PELE_XY_MODEL_ONLINE_H
+
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+#include "pele/base_potential_online.h"
+
+namespace pele {
+
+/**
+ * Adjacency list graph implementation according to S. S. Skiena,
+ * Algorithm Design Manual 
+ */
+
+struct edgenode {
+    const unsigned int y;
+    edgenode* next;
+    edgenode(const unsigned int y_, edgenode*const& next_)
+        : y(y_),
+          next(next_)
+    {}
+};
+
+class graph {
```

Comment:
There is already a graph implementation in pele.  https://github.com/pele-python/pele/blob/master/source/pele/graph.hpp 

The two implementations are optimized for different operations, so it's alright to keep both.  But probably to be clear, you should rename all the parts here adjacency_list_graph, and adjacency_list_edge_node, etc.